app/src/main/kotlin/com/intive/picover/profile/view/ProfileScreen.kt

```diff
@@ -199,23 +199,36 @@ fun ProfileScreen(viewModel: ProfileViewModel) {
 									keyboardController?.hide()
 								}
 							},
+							enabled = viewModel.isNameValidated(userName),
 						) {
 							Icon(Icons.Rounded.Check, contentDescription = null)
 						}
 					},
 				)
 			}
-			// TODO NAN add validation for text field
-			TextField(
+			Column(
 				modifier = Modifier
 					.padding(start = 20.dp, end = 20.dp, bottom = 42.dp)
 					.fillMaxWidth(),
-				value = userName,
-				onValueChange = {
-					userName = it
-				},
-				label = { Text(stringResource(id = R.string.UserName)) },
-			)
+			) {
+				TextField(
+					modifier = Modifier
+						.fillMaxWidth(),
+					value = userName,
+					onValueChange = {
+						userName = it
+					},
+					isError = !viewModel.isNameValidated(userName),
+					label = { Text(stringResource(id = R.string.UserName)) },
+				)
+				if (!viewModel.isNameValidated(userName)) {
```

Comment:
don't we have `supportingText` parameter for the material 3 textfield :D ? maybe this can be used, with error color, instead of additional Text()

Reply:
Done 😄